spec/unit/typesafe/config/path_spec.rb

```diff
@@ -0,0 +1,259 @@
+      expect("Path(foo.bar)").to eq(TestUtils.path("foo", "bar").to_s)
+      expect('Path(foo."bar*")').to eq(TestUtils.path("foo", "bar*").to_s)
+      expect('Path("foo.bar")').to eq(TestUtils.path("foo.bar").to_s)
+    end
+  end
+
+  ####################
+  # Render
+  ####################
+  context "testing .render" do
+    context "rendering simple one element case" do
+      let(:expected) { "foo" }
+      let(:path) { TestUtils.path("foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering simple two element case" do
+      let(:expected) { "foo.bar" }
+      let(:path) { TestUtils.path("foo", "bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering non safe char in an element" do
+      let(:expected) { 'foo."bar*"' }
+      let(:path) { TestUtils.path("foo", "bar*") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering period in an element" do
+      let(:expected) { '"foo.bar"' }
+      let(:path) { TestUtils.path("foo.bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering hyphen in element" do
+      let(:expected) { "foo-bar" }
+      let(:path) { TestUtils.path("foo-bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering hyphen in element" do
+      let(:expected) { "foo_bar" }
+      let(:path) { TestUtils.path("foo_bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element starting with a hyphen" do
+      let(:expected) { "-foo" }
+      let(:path) { TestUtils.path("-foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element starting with a number" do
+      let(:expected) { "10foo" }
+      let(:path) { TestUtils.path("10foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering empty elements" do
+      let(:expected) { '"".""' }
+      let(:path) { TestUtils.path("", "") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element with internal space" do
+      let(:expected) { '"foo bar"' }
+      let(:path) { TestUtils.path("foo bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering leading and trailing spaces" do
+      let(:expected) { '" foo "' }
+      let(:path) { TestUtils.path(" foo ") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering trailing space only" do
+      let(:expected) { '"foo "' }
+      let(:path) { TestUtils.path("foo ") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering number with decimal point" do
+      let(:expected) { "1.2" }
+      let(:path) { TestUtils.path("1", "2") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering number with multiple decimal points" do
+      let(:expected) { "1.2.3.4" }
+      let(:path) { TestUtils.path("1", "2", "3", "4") }
+      include_examples "path_render_test"
+    end
+  end
```

Comment:
I have no clue what is correct/conventional in rspec, but would it be a good idea to just write a helper method that takes the expected string and path (or a list of strings from which to create a path) and then using that helper in these test cases, since they're all identical?